tests/integration/config-capture-introspection-queries.test.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const { executeQuery } = require('../test-client')
+const { setupEnvConfig } = require('../agent-testing')
+const { setupApolloServerTests } = require('../apollo-server-setup')
+const queries = [
+  `{
+    __schema {
+      queryType {
+        fields {
+          name
+        }
+      }
+    }
+  }`,
+  `query introspectionType {
+    __type(name: "Library") {
+      fields {
+        name
+      }
+    }
+  }`
+]
+
+setupApolloServerTests({
+  suiteName: 'default',
+  createTests: createCaptureIntrospectionTests.bind(null, false)
+})
+
+setupApolloServerTests({
+  suiteName: 'captureIntrospectionQueries: true',
+  createTests: createCaptureIntrospectionTests.bind(null, false),
+  pluginConfig: {
+    captureIntrospectionQueries: true
+  }
+})
+
+setupApolloServerTests({
+  suiteName: 'captureIntrospectionQueries: false',
+  createTests: createCaptureIntrospectionTests.bind(null, true),
+  pluginConfig: {
+    captureIntrospectionQueries: false
+  }
+})
+
+
+function createCaptureIntrospectionTests(ignore, t) {
+  setupEnvConfig(t)
+
+  queries.forEach((query) => {
+    t.test(`should ${ignore ? '' : 'not '}ignore transaction when
+captureIntrospectionQuery is ${!ignore} and query contains
```

Comment:
Some weird indentation here but we can fix later.

Reply:
This is because of the line length. I guess I could've use ' based string but template strings mess with the rendering of the indentations. I also think we should enforce this stuff with prettier as it auto fixes style things like this